x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -0,0 +1,238 @@
+
+    public EnrichPolicyRunner(ClusterService clusterService, Client client, EnrichStore enrichStore,
+                              IndexNameExpressionResolver indexNameExpressionResolver,
+                              LongSupplier nowSupplier) {
+        this.clusterService = clusterService;
+        this.client = client;
+        this.enrichStore = enrichStore;
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+        this.nowSupplier = nowSupplier;
+    }
+
+    public void runPolicy(String policyId, ActionListener<EnrichPolicyResult> listener) throws Exception {
+        // Look up policy in policy store and execute it
+        runPolicy(policyId, enrichStore.getPolicy(policyId), listener);
+    }
+
+    public void runPolicy(final String policyName, final EnrichPolicy policy,
+                          final ActionListener<EnrichPolicyResult> listener) throws Exception {
+        // TODO: Index Pattern expansion and Alias support
+        // Collect the source index information
+        logger.info("Policy [{}]: Running enrich policy", policyName);
+        final String sourceIndexPattern = policy.getIndexPattern();
+        logger.debug("Policy [{}]: Checking source index [{}]", policyName, sourceIndexPattern);
+        GetIndexRequest getIndexRequest = new GetIndexRequest().indices(sourceIndexPattern);
+        client.admin().indices().getIndex(getIndexRequest, new ActionListener<GetIndexResponse>() {
+            @Override
+            public void onResponse(GetIndexResponse getIndexResponse) {
+                validateMappings(getIndexResponse, sourceIndexPattern, policyName, policy, listener);
+                prepareAndCreateEnrichIndex(getIndexResponse, sourceIndexPattern, policyName, policy, listener);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                listener.onFailure(e);
+            }
+        });
+    }
+
+    private Map<String, Object> getMappings(GetIndexResponse getIndexResponse, String sourceIndexName) {
+        ImmutableOpenMap<String, ImmutableOpenMap<String, MappingMetaData>> mappings = getIndexResponse.mappings();
+        ImmutableOpenMap<String, MappingMetaData> indexMapping = mappings.get(sourceIndexName);
+        // Assuming single mapping
+        MappingMetaData typeMapping = indexMapping.iterator().next().value;
+        Map<String, Object> mapping = typeMapping.sourceAsMap();
+        return mapping;
+    }
+
+    private void validateMappings(GetIndexResponse getIndexResponse, String sourceIndexPattern, String policyName, EnrichPolicy policy,
+                                  ActionListener<?> listener) {
+        logger.debug("Policy [{}]: Validating [{}] source mappings", policyName, sourceIndexPattern);
+        Map<String, Object> mapping = getMappings(getIndexResponse, sourceIndexPattern);
+        @SuppressWarnings("unchecked")
+        Map<String, Object> properties = ((Map<String, Object>) mapping.get("properties"));
```

Comment:
maybe `Map<?, ?> properties` in order to remove the suppress warning annotation?